tests/Unit/TestCreation.hpp

```diff
@@ -3,50 +3,99 @@
 
 #pragma once
 
+#include <boost/algorithm/string.hpp>
 #include <memory>
 #include <string>
 
 #include "Options/Options.hpp"
 #include "Options/ParseOptions.hpp"
 #include "Utilities/NoSuchType.hpp"
 
+namespace TestHelpers {
 namespace TestCreation_detail {
+template <typename Tag, typename = void>
+struct AddGroups {
+  template <typename OptionTag = Tag>
+  static std::string apply(std::string construction_string) noexcept {
+    construction_string.insert(0, 2, ' ');
+    construction_string =
+        boost::algorithm::replace_all_copy(construction_string, "\n", "\n  ");
+    construction_string.insert(
+        0, Options_detail::name_helper<OptionTag>::name() + ":\n");
+    return construction_string;
+  }
+};
+
+template <typename Tag>
+struct AddGroups<Tag, cpp17::void_t<typename Tag::group>> {
+  static std::string apply(const std::string& construction_string) noexcept {
+    return AddGroups<typename Tag::group>::apply(
+        AddGroups<void>::template apply<Tag>(construction_string));
+  }
+};
+}  // namespace TestCreation_detail
+
+/// \ingroup TestingFrameworkGroup
+/// The default option tag for `TestHelpers::test_creation()`,
+/// `TestHelpers::test_factory_creation()`, and
+/// `TestHelpers::test_enum_creation()`.
 template <typename T>
-struct Opt {
+struct TestCreationOpt {
   using type = T;
   static constexpr OptionString help = {"halp"};
 };
-}  // namespace TestCreation_detail
 
 /// \ingroup TestingFrameworkGroup
-/// Construct an object from a given string.  Each line in the string
-/// must be indented.
-template <typename T, typename Metavariables = NoSuchType>
+/// Construct an object from a given string. The string must not contain the tag
+/// name.
```

Comment:
Can you clarify what "the tag" is? Do you perhaps mean "the class name"? E.g. in the test, you create a `ClassWithoutMetavariables` but there is no "Tag" (in the Databox sense) for this class.

Perhaps making a Doxygen snippet from the tests would be useful here. Alternatively, you could note in the documentation that the test gives many examples.

This comment also applies to `test_factory_creation` and `test_enum_creation`.